portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```
@@ -41,8 +41,9 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override 
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String comment = getParameter(request, "comment", "");
-    if (!comment.equals("")) {
-      commentHandler.addCommentToDatabase(comment);
+    UserManager userManager = new UsersApiUserManager();
+    if (!comment.equals("") && userManager.userIsLoggedIn()) {
```

you don't have to do it on this project but in the future, if the input is wrong - empty comment - it's better to reject the request instead of silently ignore it.
Also the UI should not send the request if the comment is empty.